src/codegen/codegen_aocl.cc

```diff
@@ -31,17 +31,16 @@ runtime::Module BuildAOCL(Array<LoweredFunc> funcs, std::string target_str) {
   runtime::SaveBinaryToFile("aocl.cl", code.c_str());
 
   // Compile the .cl file.
+  std::string cmd = "aoc aocl.cl";
   Target target = Target::create(target_str);
-  if (target->device_name == "") {
-    LOG(FATAL) << "AOCL device name not specified in build target.";
+  if (target->device_name != "") {
+    cmd += " -board=" + target->device_name;
   }
-  std::string cmd = "aoc aocl.cl";
   for (std::string option : target->options()) {
     if (option == "-mattr=emulator") {
       cmd += " -march=emulator";
     }
   }
-  cmd += " -board=" + target->device_name;
```

Comment:
Removing target device name look good to me.
'aoc' uses 'AOCL_BOARD_PACKAGE_ROOT' environment variable to determine the board name when '-board=' is not specified.